### PR TITLE
Add additional SSL tests from ponyc

### PR DESCRIPTION
**Pull request by SeanTAllen**
Missed this on the first setup.

### CODE REVIEW

**Comment by SeanTAllen** [comment]
@Theodus I'm leaving style guide fixes for another PR.

File: examples/simple-example/simple-example.pony [hunk 1]
```diff
@@ -8,12 +8,21 @@ actor Main
...
-        return
+    try
+      let auth = env.root as AmbientAuth
+      let sslctx: (SSLContext | None) =
+        try
+          recover
+            // paths need to be adjusted to a absolute location or you need to run the example
+            // from a location where this relative path will be valid
+            SSLContext
+              .>set_authority(FilePath(auth, "assets/cert.pem")?)?
```
Comment (Theodus) [inline examples/simple-example/simple-example.pony]: There should be a space following the `.>` operators.

File: net_ssl/_test.pony [hunk 2]
```diff
@@ -114,3 +123,524 @@ class iso _TestALPNStandardProtocolResolver is UnitTest
...
+                FilePath(auth, "assets/key.pem")?)?
+            .> set_client_verify(true)
+            .> set_server_verify(true)
+        end
+      else
+        h.fail("set_cert failed")
+        error
+      end
+
+    let ssl_client = try sslctx.client()? else h.fail("failed getting ssl client session"); error end
```
Comment (Theodus) [inline net_ssl/_test.pony]: I think these lines extend past the 80 column limit

**Comment by SeanTAllen** [comment]
@Theodus updated. let me know if good, once it is, i can squash and merge.

File: net_ssl/_test.pony [hunk 3]
```diff
@@ -640,7 +640,17 @@ primitive _TestSSLContext
         error
       end
 
-    let ssl_client = try sslctx.client()? else h.fail("failed getting ssl client session"); error end
-    let ssl_server = try sslctx.server()? else h.fail("failed getting ssl server session"); error end
+    let ssl_client = try
```
Comment (Theodus) [inline net_ssl/_test.pony]: The try block should start on the next line, indented one level

File: net_ssl/_test.pony [hunk 4]
```diff
@@ -640,7 +640,17 @@ primitive _TestSSLContext
...
+      sslctx.client()?
+    else
+      h.fail("failed getting ssl client session")
+      error
+    end
+    let ssl_server = try
```
Comment (Theodus) [inline net_ssl/_test.pony]: Here as well

File: net_ssl/_test.pony (hunk 3 above)
Comment (SeanTAllen) [reply net_ssl/_test.pony]: is there any indentation of the `try` when it is on the next line?

Comment (Theodus) [reply net_ssl/_test.pony]: yes: https://github.com/ponylang/ponyc/blob/master/STYLE_GUIDE.md#assignment

**Pull request merged**